sirplatform.h

```diff
@@ -124,13 +124,6 @@
 
 # define SIR_MAXHOST 256
 
-# if defined(__GLIBC__)
-#  if (__GLIBC__ >= 2 && __GLIBC_MINOR__ > 19) || \
-     ((__GLIBC__ == 2 && __GLIBC_MINOR__ <= 19) && defined(_BSD_SOURCE))
-#   define __HAVE_UNISTD_READLINK__
-#  endif
-# endif
-
```

Comment:
readlink is required by POSIX (it's part of POSIX.1-2008) and it's present on essentially everything going back to 2002, so I can't imagine why it would ever be hidden, and it isn't on any of the systems I've tested.  And on most other non-glibc Linux systems there are no such "feature tests" (since it's a required part of POSIX) so that ifdef block just grows to have to include OS detection for every OS we support.

Reply:
Also, I'm not too worried about glibc 2.18 and earlier.

Ubuntu 14.04 has 2.19, and that is the earliest version the repos are available for.

RHEL 7 has a glibc that is based on 2.17 - but is EOL in 10 months.  I can check it if need be.  There are no other maintained distributions that use such a version I can identify.